dia-batching-server/src/api/polygon.rs

```diff
@@ -0,0 +1,273 @@
+use crate::api::error::PolygonError;
+use crate::api::Quotation;
+use crate::args::PolygonConfig;
+use crate::AssetSpecifier;
+use rust_decimal::Decimal;
+use serde::de::DeserializeOwned;
+use serde::{Deserialize, Serialize};
+
+pub struct PolygonPriceApi {
+	client: PolygonClient,
+}
+
+impl PolygonPriceApi {
+	pub fn new_from_config(config: PolygonConfig) -> Self {
+		let api_key = config.pg_api_key.expect("Please provide a Polygon API key");
+
+		Self::new(config.pg_host_url, api_key)
+	}
+
+	pub fn new(host_url: String, api_key: String) -> Self {
+		let client = PolygonClient::new(host_url, api_key);
+
+		Self { client }
+	}
+	pub async fn get_prices(
+		&self,
+		assets: Vec<&AssetSpecifier>,
+	) -> Result<Vec<Quotation>, PolygonError> {
+		let asset_and_source_currency = assets
+			.into_iter()
+			.filter_map(|asset| {
+				let source_currency = PolygonPriceApi::extract_source_currency(asset);
+				match source_currency {
+					Some(polygon_id) => Some((asset, polygon_id)),
+					None => {
+						log::warn!("Unsupported polygon asset: {:?}", asset);
+						None
+					},
+				}
+			})
+			.collect::<Vec<_>>();
+
+		let mut prices = Vec::new();
+		for (asset, source_currency) in asset_and_source_currency {
```

Comment:
I think it should be fine. We have an unlimited number of queries available in our plan. Though you do have a point that it might be better to batch the requests with that 'All tickers' endpoint. 

I was curious how much the delay is and one query takes approximately 0.5 seconds on my machine. If we want to support more currencies in the future or run the server on a lower-spec machine, the delay could add up to an undesirable amount. 
```
 INFO  dia_batching_server::api::polygon > Time before fetching prices: 2024-07-02 16:05:05.065909 UTC
 INFO  dia_batching_server::api::polygon > Adding price for TZS-USD at time 2024-07-02 16:05:05.653694 UTC
 INFO  dia_batching_server::api::polygon > Adding price for NGN-USD at time 2024-07-02 16:05:06.276484 UTC
 INFO  dia_batching_server::api::polygon > Adding price for BRL-USD at time 2024-07-02 16:05:06.752161 UTC
 INFO  dia_batching_server::api::polygon > Adding price for AUD-USD at time 2024-07-02 16:05:07.246598 UTC
 INFO  dia_batching_server::api::polygon > Adding price for EUR-USD at time 2024-07-02 16:05:07.842225 UTC
 INFO  dia_batching_server::api::polygon > Time after fetching prices: 2024-07-02 16:05:07.842358 UTC
```

So I think you have a point and I'll try to switch to that other endpoint instead.

Reply:
Wow 0.5 seconds seems quite a lot, thanks for testing it. That would mean that supporting 24 currencies would be already too much for a block "resolution". 

But since we are not supporting that many now, maybe you can leave a comment and we can pick this up later as an improvement since it will change the logic a bit, I assume.

Reply:
I added that batch request now because the results also give a higher precision than just 5 decimals. 

I used the price contained in the `bid` of the last quote because when comparing the price returned by the fiat-conversion endpoint to the tickers, it seemed to match with the value of the `bid` but rounded.